lib/models/addon-discovery.js

```diff
@@ -20,241 +20,238 @@ var findup     = require('findup');
   @extends CoreObject
   @constructor
 */
-
-function AddonDiscovery(ui) {
-  this.ui = ui;
-}
-
-AddonDiscovery.__proto__ = CoreObject;
-AddonDiscovery.prototype.constructor = AddonDiscovery;
-
-/**
-  This is one of the primary APIs for this class and is called by the project.
-  It returns a tree of plain objects that each contain information about a
-  discovered addon. Each node has `name`, `path`, `pkg` and
-  `childAddons` properties. The latter is an array containing any addons
-  discovered from applying the discovery process to that addon.
-
-  @private
-  @method discoverProjectAddons
- */
-AddonDiscovery.prototype.discoverProjectAddons = function(project) {
-  var projectAsAddon = this.discoverFromProjectItself(project);
-  var internalAddons = this.discoverFromInternalProjectAddons(project);
-  var cliAddons = this.discoverFromCli(project.cli);
-  var dependencyAddons;
-
-  if (project.hasDependencies()) {
-    dependencyAddons = this.discoverFromDependencies(project.root, project.nodeModulesPath, project.pkg, false);
-  }  else {
-    dependencyAddons = [];
-  }
-
-  var inRepoAddons = this.discoverInRepoAddons(project.root, project.pkg);
-  var addons = projectAsAddon.concat(cliAddons, internalAddons, dependencyAddons, inRepoAddons);
-
-  return addons;
-};
-
-/**
-  This is one of the primary APIs for this class and is called by addons.
-  It returns a tree of plain objects that each contain information about a
-  discovered addon. Each node has `name`, `path`, `pkg` and
-  `childAddons` properties. The latter is an array containing any addons
-  discovered from applying the discovery process to that addon.
-
-  @private
-  @method discoverProjectAddons
- */
-AddonDiscovery.prototype.discoverChildAddons = function(addon) {
-  debug('discoverChildAddons: %s(%s)', addon.name, addon.root);
-  var dependencyAddons = this.discoverFromDependencies(addon.root, addon.nodeModulesPath, addon.pkg, true);
-  var inRepoAddons = this.discoverInRepoAddons(addon.root, addon.pkg);
-  var addons = dependencyAddons.concat(inRepoAddons);
-
-  addons = addons.filter(function(childAddon) {
-    return !addon.shouldIncludeChildAddon || addon.shouldIncludeChildAddon(childAddon);
-  });
-
-  return addons;
-};
-
-/**
-  Returns an array containing zero or one nodes, depending on whether or not
-  the passed project is an addon.
-
-  @private
-  @method discoverFromProjectItself
- */
-AddonDiscovery.prototype.discoverFromProjectItself = function(project) {
-  if (project.isEmberCLIAddon()) {
-    var addonPkg = this.discoverAtPath(project.root);
-    if (addonPkg) {
-      return [addonPkg];
+var AddonDiscovery = CoreObject.extend({
+  init: function(ui) {
+    this._super.init.apply(this, arguments);
```

Comment:
Should be `this._super.apply(this, arguments);` (emulating ES6 classes where `super()` is used in constructor, but `super.someMethod()` is used everywhere else).